### PR TITLE
IGP-29

### PR DESCRIPTION
**Pull request by thrilok209**

### CODE REVIEW

File: contracts/payloads/IGP29/PayloadIGP29.sol [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+
+    address public constant ETH_ADDRESS =
+        0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    address public constant wstETH_ADDRESS =
+        0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0;
+    address public constant weETH_ADDRESS =
+        0xCd5fE23C85820F7B72D0926FC9b05b43E359b7ee;
+
+    address public constant PT_sUSDe_ADDRESS =
+        0x9D39A5DE30e57443BfF2A8307A4256c8797A3497; // TODO
```
Comment (bergben) [inline contracts/payloads/IGP29/PayloadIGP29.sol:336]: 0x6c9f097e044506712B58EAC670c9a5fd4BCceF13

File: contracts/payloads/IGP29/PayloadIGP29.sol [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+
+    function verifyProposal() external view {}
+
+    /***********************************|
+    |     Proposal Payload Actions      |
+    |__________________________________*/
+
+    /// @notice Action 1: Set PT_sUSDe token config and market rate curve on liquidity.
+    function action1() internal {
+        {
```
Comment (bergben) [inline contracts/payloads/IGP29/PayloadIGP29.sol:400]: other way aroun., rate must be set before token config otherwise this will revert

Comment (thrilok209) [reply contracts/payloads/IGP29/PayloadIGP29.sol:400]: Yess! Fixed it during simulation.

**Review by bergben — APPROVED** [review]

File: contracts/payloads/IGP29/PayloadIGP29.sol [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+        require(address(this) == address(TIMELOCK), "not-valid-caller");
+
+        /// Action 1: Approve fUSDC and fUSDT protocols to spend the reserves tokens
+        action1();
+
+       /// Action 2: Add config handler on liquidity layer for sUSDe/USDC & sUSDe/USDT vault.
+       action2();
+
+       /// Action 3: call cast() - transfer 100 stETH to Team Multisig from treasury.
+        action3();
```
Comment (KaymasJain) [inline contracts/payloads/IGP29/PayloadIGP29.sol:403]: We can also transfer ETH from reserve contract itself?

File: contracts/payloads/IGP29/PayloadIGP29.sol [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+            addr: address(VAULT_sUSDe_USDC_CONFIG_HANDLER),
+            value: true
+        });
+
+        configs_[1] = AdminModuleStructs.AddressBool({
+            addr: address(VAULT_sUSDe_USDT_CONFIG_HANDLER),
+            value: true
+        });
+
+        LIQUIDITY.updateAuths(configs_);
```
Comment (bergben) [inline contracts/payloads/IGP29/PayloadIGP29.sol:449]: These must be set at VaultFactory as vault auth, see https://github.com/Instadapp/inst-governance/blob/main/contracts/payloads/IGP15/PayloadIGP15.sol#L661

Comment (thrilok209) [reply contracts/payloads/IGP29/PayloadIGP29.sol:449]: Okay!

Comment (bergben) [reply contracts/payloads/IGP29/PayloadIGP29.sol:449]: > Same as on [IGP15](https://github.com/Instadapp/inst-governance/blob/main/contracts/payloads/IGP15/PayloadIGP15.sol#L661), for the sUSDe v1.1.0 vaults we must set the ethena rate config handlers as vault auths at the Vault factory:
 Vault_sUSDe_USDC 0x3996464c0fCCa8183e13ea5E5e74375e2c8744Dd, config handler 0xa7C805988f04f0e841504761E5aa8387600e430b
 Vault_sUSDe_USDT 0xBc345229C1b52e4c30530C614BB487323BA38Da5, config handler 0x7607968F40d7Ac4Ef39E809F29fADDe34C00A0A6
These two handlers we will also have to add to config rebalance triggers from API if not present there yet (

Comment (thrilok209) [reply contracts/payloads/IGP29/PayloadIGP29.sol]: image

**Review by bergben — APPROVED** [review]

**Pull request merged**